pilot/pkg/security/authz/model/generator.go

```diff
@@ -191,7 +191,34 @@ func (requestHeaderGenerator) permission(key, value string, forTCP bool) (*rbacp
 	if err != nil {
 		return nil, err
 	}
+
+	// When the port is 80 and the scheme is http, authority header does not contain a port. The situation is
```

Comment:
Do we want to change the policy or change the header in requests? To always include the port

Reply:
My thought is I don't know if we want to always include the port in the authority header, as it may cause unintended consequences for anything else that utilizes that header. It would definitely simplify the generated config

Reply:
Assuming its feasible in Envoy to always add it that seems preferred. I can't imagine anything being upset that the port is there ?

This is HBONE traffic too, not user originated, so we can be stricter on what we accept.

Reply:
I didn't see a way to manipulate that without a WASM plugin. But I am also far from an Envoy expert. I'll dig through the envoy docs a bit more and see what I can find as I agree that would probably be the better approach.

Reply:
@kyessenov could use your expertise here?

Reply:
Another alternative: instead of matching HBONE properties can we somehow turn those into the L4 attributes (e.g. we check the dest port as normal instead of `:authority`)?

Reply:
It's incorrect to use L4 attributes for authority matching, https://www.ietf.org/archive/id/draft-ietf-httpbis-semantics-13.html#section-4.3.3-6. I think having a port-qualified authority as a basic attribute seems like the cleanest option.  We could probably extend the RBAC matcher to support that if it doesn't already.

Reply:
The issue is:

* Outer listener: We get HBONE request with `:authority=1.2.3.4:999`. Send to internal listener for 1.2.3.4:999.
* Inner listener: we want to implement a "port match". If we did this naively, the dest port  would be 15008, we want to treat it as `999`. To workaround this we match the `:authority` header from the outer listener

Reply:
I think the easiest (and probably required) thing is to make ztunnel always qualify the authority for 80 and 443 in HBONE. Is there a problem with that approach?

Reply:
Per HTTP spec I mentioned, L4 port should not be used for secure HTTP transport. The idea behind that is that UDS is secure and port-less. gRPC, for example, always requires fully qualified 80 and 443 ports.

Reply:
Ztunel does, but Envoy doesn't and we can get HBONE requests from Envoy. I didn't see any obvious way to include port 80/443 in envoy.

Reply:
Ok, so let's fully qualify authority in the envoy upstream. Can you send me a pointer to code where it's difficult?

Reply:
I haven't found a way to fully qualify  the authority header. Been digging through docs and haven't found anything so far. I see some PR comments that mention the behavior of stripping ports that match the scheme from the authority header. Still digging through, however.

Reply:
I think its actually just gateway that has an issue... seems to work for most other cases, might be an issue with our config on gateway specifically. will investigate.

Reply:
Ok so this was a lot of misunderstanding.

Previously the idea is we had this RBAC transform from a port match to an authority suffix match (https://github.com/istio/istio/pull/41491). I don't know if this was just flawed from the beginning, broken at some point, or meant for the envoy-based ztunnel, but it ended up being applied in the internal pod listener.

So if I send a request like `curl 10.244.0.28 -H "Host: 10.244.0.28:81` That rbac filter will see something like this:



```
checking request: requestedServerName: , sourceIP: envoy://internal_client_address/, directRemoteIP: envoy://internal_client_address/, remoteIP: 10.244.0.1:0,localAddress: envoy://inbound-pod|80||10.244.0.28/, ssl: none, headers: ':authority', '10.244.0.28:81'
```

If my policy is "Allow port 81", this is obviously broken and user bypassed. Also probably entirely broken for TCP traffic.

The alternative, though, is also broken. We cannot just match "dest port = 80", because the destination is actually `envoy://inbound-pod|80||10.244.0.28/` which has no port. Logically we know it means port 80, but Envoy doesn't.

So I think the behavior is broken with and without this PR, but I am not sure how to fix it. Perhaps we don't, since it sounds like we might kill this internal pod listener soon?

Reply:
From the way you describe it, we always know the pod port from the pod listener name, so we can compute the RBAC answer ahead of time.

Reply:
Ah of course... thanks!! So we can just convert the port match into either "always match" or "never match" (technically we could optimize it more, but probably not worth it in the short term).

Eventually we would want arbitrary ports on inbound pod, but we can handle that when we remove the internal listener IMO

Reply:
Opened revert commit and will close this PR.